blog/2023-02-20-introducing-the-dhis2-java-sdk.md

```diff
@@ -0,0 +1,174 @@
+:::
+
+The `sourceDhis2Client` method chain retrieving the data value set returns an object belonging to a `DataValueSet` _resource class_. Rather than attempting to parse raw JSON strings or reference unsafe generic _Maps_, a resource class such as `DataValueSet` provides a type-safe fluent view of the request/response's content to the application developer. Additionally, with the help of an IDE’s autocompletion, the programmer can explore which fields are available instead of digging into the Web API documentation. 
+
+Resource classes are located within the package `org.hisp.dhis.api.model.vX_X_X` where `X_X_X` is a variable informing us of the DHIS2 version that the classes are compatible with. Taking `DataValueSet` from our example, this class is located inside the package `org.hisp.dhis.api.model.v2_39_1`. The version no. `v2_39_1` means that this `DataValueSet` class is compatible with DHIS 2.39.1. If we decide to update the application to support a newer version of DHIS2, then we should bump the package version no. `v2_39_1` to match the new DHIS2 version. Any Web API breaking changes like the renaming of a JSON property would be reflected in the corresponding resource class. This in turn would cause the application to fail compilation should it be calling an accessor that is impacted by the breaking change. 
+
+:::info
+`org.hisp.dhis.api.model.vX_X_X` packages shipped with the SDK correspond to the last 3 versions of DHIS2 at the time the SDK was released.
+:::
+
+Despite the type-safety resource classes offer, the SDK doesn’t have an opinionated way of representing resources. One can always obtain the raw JSON like so:
+
+```java
+String dataValueSetAsJson = sourceDhis2Client.get( "dataValueSets" )
+    .withParameter( "dataSet", "BfMAe6Itzgt" )
+    .withParameter( "period", "202302" )
+    .withParameter( "orgUnit", "PLq9sJluXvc" )
+    .transfer()
+    .returnAs( String.class );
+```
+
+Another alternative is to deserialise the JSON response into a _Map_:
+
+```java
+Map<String, Object> dataValueSet = sourceDhis2Client.get( "dataValueSets" )
+    .withParameter( "dataSet", "BfMAe6Itzgt" )
+    .withParameter( "period", "202302" )
+    .withParameter( "orgUnit", "PLq9sJluXvc" )
+    .transfer()
+    .returnAs( Map.class );
+```
+
+For the memory conscious folks, so that large payloads don’t take up all the machine’s memory, the response can also be streamed with the `read()` method instead of `returnAs(...)`:
```

Comment:
```suggestion
For those who are mindful of memory usage, it's possible to stream the response using the `read()` method instead of `returnAs(...)` to avoid large payloads consuming the machine's memory, as shown in the example below:
```

Reply:
could you remove the word "folks" from this sentence?